src/clients/http_client/__tests__/http_client.test.ts

```diff
@@ -424,6 +442,14 @@ describe('HTTP client', () => {
         ),
       },
     }).toMatchMadeHttpRequest();
+  });
+
+  it('extends a User-Agent provided by config and an extra header', async () => {
```

Comment:
Split out this test because the mock adapter is not comparing headers when searching for comparison, hence was finding the first one in the pair because all the other details are the same ... could fake it by changing the URL for each test ... 🤔 but is it a huge deal if they're separate tests?

Reply:
Resolved while pairing with @paulomarg